packages/devtools_app/lib/src/framework/notifications.dart

```diff
@@ -7,84 +7,100 @@ import 'dart:async';
 import 'package:flutter/material.dart';
 import 'package:flutter/scheduler.dart';
 
+import '../primitives/auto_dispose_mixin.dart';
+import '../primitives/notifications.dart';
 import '../primitives/utils.dart';
-import 'common_widgets.dart';
-import 'theme.dart';
-import 'utils.dart';
+import '../shared/common_widgets.dart';
+import '../shared/globals.dart';
+import '../shared/theme.dart';
+import '../shared/utils.dart';
 
 double get _notificationHeight => scaleByFontFactor(175.0);
 final _notificationWidth = _notificationHeight * goldenRatio;
 
-/// Interface for pushing notifications in the app.
-///
-/// Use this interface in controllers that need to show notifications.
-///
-/// Using the interface instead of the [NotificationsState] implementation
-/// will allow you to write unit tests for the controller that consumes it
-/// instead of widget tests.
-abstract class NotificationService {
-  /// Pushes a notification [message].
-  void push(String message);
-}
-
 /// Manager for notifications in the app.
 ///
 /// Must be inside of an [Overlay].
```

Comment:
is this comment accurate? the Overlay is the top level widget in this build method, and it doesn't look like the `NotificationsView` widget is not inside an Overlay where it is used.

I'd also recommend combining the `NotificationsView` and the `_Notifications` widget since this one is so minimal and that would eliminate some unnecessary boilerplate. Especially since the `_NotificationsState` build method just returns `return widget.child;`

Reply:
I tried it and it did not work for some reasons. I will postpone it as I want to commit this PR and switch to next tasks. Left the comment.